source/slang/slang-type-layout.cpp

```diff
@@ -763,11 +768,16 @@ struct CUDAObjectLayoutRulesImpl : CPUObjectLayoutRulesImpl
                 // TODO(JS): We are storing as a pointer for now
 
                 // It's a pointer and a size in bytes
-                return SimpleLayoutInfo(LayoutResourceKind::Uniform, sizeof(void*), sizeof(void*));
+                return SimpleLayoutInfo(LayoutResourceKind::Uniform, sizeof(void*), SLANG_ALIGN_OF(void*));
 
             case ShaderParameterKind::SamplerState:
-                // It's a pointer
-                return SimpleLayoutInfo(LayoutResourceKind::Uniform, sizeof(void*), sizeof(void*));
+                // In CUDA it seems that sampler states are combined into texture objects.
+                // So it's a binding issue to combine a sampler with a texture - and sampler are ignored
+                // For simplicity here though - we do create a variable and that variable takes up
+                // uniform binding space.
+                // TODO(JS): If we wanted to remove these variables we'd want to do it as a pass. The pass
```

Comment:
This comment is reasonable, but in practice I might argue for a pass that replaces `SamplerState` types with `void`, and then we should have a general-purpose big-hammer pass that rewrites any empty types with `void` and erases any `void`-type values from existence in the code (e.g., eliminate `void`-type arguments, parameters, locals, etc.)